internal/pkg/plugin/definition.go

```diff
@@ -53,14 +55,16 @@ type Config map[string]interface{}
 // GetPluginName returns a plugin name generated by the packages.
 // Recommended plugin directory structure: plugins/{plugin-category}/{plugin-name}, so the generated names is {plugin-name}-{plugin-category}.
 func GetPluginName(p Plugin) string {
-	nameParts := 2
 	t := reflect.TypeOf(p)
 	if t.Kind() == reflect.Ptr {
 		t = t.Elem()
 	}
-	arr := strings.Split(t.PkgPath(), "/")
-	if len(arr) < nameParts {
-		return ""
+	path := t.PkgPath()
+	path = path[strings.LastIndex(path, PluginDir)+len(PluginDir)+1:]
+	arr := strings.Split(path, "/")
+	if len(arr) > 1 {
+		category := arr[0]
+		return strings.Join(arr[1:], "-") + "-" + category
```

Comment:
The first feeling is so complicated.
`plugin` is an interface, why not organize name in the specific plugin implementation. And then we can use `plugin.Name()` here instead of L62~67.

Reply:
The directory structure is an effective way to avoid duplication of plug-in names, and we recommend this util, but it is not mandatory

Reply:
@wu-sheng @surlymo  what do u think about it ?

Reply:
As you asked, personally, prefer to use `plugin.Name()` to simplify.